File: 268.missingNumber.py
#
# @lc app=leetcode id=268 lang=python3
#
# [268] Missing Number
#

# @lc code=start
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        length = len(nums)
        actual_sum = (length * (length+1))// 2
        array_sum = 0
        for num in nums:
            array_sum += num
        return actual_sum- array_sum
        
# @lc code=end

File: 268.missingNumber.js
/*
 * @lc app=leetcode id=268 lang=javascript
 *
 * [268] Missing Number
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
  const len = nums.length;
  const actualSum = (len * (len + 1)) / 2;
  let arraySum = 0;
  //   nums.map((num) => arraySum + num);
  nums.forEach((num) => {
    arraySum = arraySum + num;
  });
  console.log(arraySum, actualSum);
  return actualSum - arraySum;
};
// @lc code=end
